save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/DownloadFilesController.kt

```diff
@@ -162,23 +161,18 @@ class DownloadFilesController(
         )
     )
 
-    private fun doDownload(fileKey: FileKey): Mono<ByteBufferFluxResponse> = Mono.fromCallable {
-        logger.info("Sending file ${fileKey.name} to a client")
-        val content = fileStorage.download(fileKey)
-        ResponseEntity.ok()
-            .contentType(MediaType.APPLICATION_OCTET_STREAM)
-            .body(content)
-    }
-        .doOnError(FileNotFoundException::class.java) {
-            logger.warn("File with key $fileKey is not found", it)
+    private fun doDownload(fileKey: FileKey): Mono<ByteBufferFluxResponse> = fileStorage.doesExist(fileKey)
+        .filter { it }
+        .switchIfEmptyToNotFound {
+            "File with key $fileKey is not found"
         }
-        .onErrorReturn(
-            FileNotFoundException::class.java,
-            ResponseEntity
-                .status(HttpStatus.NOT_FOUND)
+        .map {
+            logger.info("Sending file ${fileKey.name} to a client")
+            val content = fileStorage.download(fileKey)
```

Comment:
content is used once? - may be move it directly to a `.body(fileStorage.download(fileKey))`

Reply:
done